typescript/packages/subsurface-viewer/src/components/Map.tsx

```diff
@@ -987,22 +989,43 @@ export function jsonToObject(
     return jsonConverter.convert(filtered_data);
 }
 
+/**
+ * Creates layers according to layer definiton Records.
+ * Keeps the relative order of layers.
+ * @param layers Array of layer definition Records or externally created layers.
+ * @returns Array of layer instances.
+ */
 export function createLayers(
-    data: Record<string, unknown>[],
+    layers: TLayerDefinition[],
     enums: Record<string, unknown>[] | undefined = undefined
 ): LayersList {
     const configuration = createConfiguration(enums);
-    const layersList: Layer[] = [];
 
-    // remove empty data/layer object
-    const filtered_data = data.filter((value) => !isEmpty(value));
-    for (const layerData of filtered_data) {
-        const layer = createLayer(layerData, configuration);
-        if (layer) {
-            layersList.push(layer);
+    // remove empty record/layer objects
+    const filtered_layers = layers.filter((value) => !isEmpty(value)) as (
+        | Record<string, unknown>
+        | Layer
+    )[];
+    const layersList = new Array(filtered_layers.length);
+    const jsonLayerIndices: number[] = [];
```

Comment:
I never like side by side arrays, whenever other solutions are possible (error prone, less readable).

can you use:
```
const jsonIndexedLayers: {layer: Record<string, unknown>, index: number}[];
...
jsonIndexedLayers.push({layer: layer, index: index});
```
But I let you choose to resolve this...